src/modules/json.wren

```diff
@@ -0,0 +1,252 @@
+import "io" for FileSystem
+
+class JsonOptions {
+    static NIL { 0 }
+    static ESCAPE_SLASHES { 1 }
+    static ABORT_ON_ERROR { 2 }
+
+    static shouldAbort(options) {
+        return (options == JsonOptions.ABORT_ON_ERROR || options == (JsonOptions.ABORT_ON_ERROR | JsonOptions.ESCAPE_SLASHES))
+    }
+}
+
+class JsonError {
+    line { _line }
+    position { _position }
+    message { _message }
+    found { _found }
+
+    construct new(line, pos, message, found) {
+        _line = line
+        _position = pos
+        _message = message
+        _found = found
+    }
+
+    static empty() {
+        return JsonError.new(0, 0, "", false)
+    }
+}
+
+class JsonStream {
+    foreign stream_begin(value)
+    foreign stream_end()
+    foreign next
+    foreign value
+    foreign error_message
+    foreign lineno
+    foreign pos
+    foreign static escapechar(value, options)
+
+    namespace { "com.domeengine.json:" }
+    isNull { namespace + "JSON_NULL" }
+    isString { namespace + "JSON_STRING" }
+    isNumeric { namespace + "JSON_NUMBER" }
+    isBoolTrue { namespace + "JSON_TRUE" }
+    isBoolFalse { namespace + "JSON_FALSE" }
+    isArray { namespace + "JSON_ARRAY" }
+    isArrayEnd { namespace + "JSON_ARRAY_END" }
+    isObject { namespace + "JSON_OBJECT" }
+    isObjectEnd { namespace + "JSON_OBJECT_END" }
+    isDone { namespace + "JSON_DONE" }
+    isError { namespace + "JSON_ERROR" }
+    isInit { namespace + "JSON_INIT" }
+
+    result { _result }
+    error { _error }
+    options { _options }
+    raw { _raw }
+
+    construct new(raw, options) {
+        _result = {}
+        _error = JsonError.empty()
+        _lastEvent = isInit
+        _raw = raw
+        _options = options
+    }
+
+    begin() {
+        stream_begin(_raw)
+        _result = process(next)
+    }
+
+    end() {
+        stream_end()
+    }
+
+    process(event) {
+
+        _lastEvent = event
+
+        if(event == isBoolTrue || event == isBoolFalse) {
```

Comment:
As a code-style thing, I prefer spaces after `if` and `while` keywords in the DOME codebase.